File: apps/api/src/main/java/br/com/agendaplatform/scheduling/AppointmentSummary.java
package br.com.agendaplatform.scheduling;

import java.time.Instant;
import java.util.UUID;

public record AppointmentSummary(
        UUID id,
        UUID clientId,
        String clientName,
        UUID serviceId,
        String serviceName,
        Instant startAt,
        Instant endAt,
        String status,
        String cancellationReason,
        UUID seriesId) {
}

File: apps/admin/src/app/core/scheduling/appointment-summary.ts
export interface AppointmentSummary {
  readonly id: string;
  readonly clientName: string;
  readonly serviceName: string;
  readonly startAt: string;
  readonly endAt: string;
  readonly status: 'SCHEDULED' | 'CONFIRMED' | 'ARRIVED' | 'IN_PROGRESS' | 'DONE' | 'CANCELLED' | 'NO_SHOW';
  readonly cancellationReason: string | null;
  readonly seriesId: string | null;
}
